woocommerce/woocommerce
# Fix transient notices overlapping product editor footer

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





In #38599, the product editor footer was introduced.

Transient notices overlapped this new footer.

This PR updates the the transient notices to not overlap the new footer.

To support this change, the `TransientNotices` component was moved from `woocommerce/client/admin` to `@woocommerce/components`, so that it could be used from both `@woocommerce/product-editor` and `woocommerce/client/admin`. This allows `TransientNotices` to be cleanly positioned in both layouts, without any funky JS/CSS positioning hacks, accommodating the variable height product editor footer.

Closes #38651.

#### Before (overlapping transient notices)

<img width="1115" alt="Screenshot 2023-06-05 at 19 36 54" src="https://github.com/woocommerce/woocommerce/assets/2098816/6421a525-fd8a-4299-936d-2f9a1a26d063">

#### After (transient notices above footer)

<img width="1097" alt="Screenshot 2023-06-09 at 15 11 31" src="https://github.com/woocommerce/woocommerce/assets/2098816/e9fa3f6e-78ac-4305-a7d2-118df08f9aa3">



### How to test the changes in this Pull Request:



Note: To reset the state of the system so that the feedback footer will be shown, delete the following options:
* `woocommerce_product_editor_show_feedback_bar`
* `woocommerce_ces_shown_for_actions`

1. Enable experimental "new product editor" (beta) in WooCommerce > Settings > Advanced > Features
2. Go to Products > Add New
3. Verify no feedback footer is shown.
4. Fill out some fields and save draft (or publish).
5. Verify feedback footer is shown and transient notice is shown correctly above it.
6. Dismiss the footer.
7. Make a change to the product and save draft.
8. Verify the transient notice is shown correct at the bottom of the screen.
